src/query/management/src/role/role_mgr.rs

```diff
@@ -247,6 +247,43 @@ impl RoleApi for RoleMgr {
         Ok(Some(seq))
     }
 
+    /// Only drop role will call transfer.
+    ///
+    /// If a role is dropped, but the owner object is exists,
+    ///
+    /// The owner role need to update to account_admin.
+    #[async_backtrace::framed]
+    #[minitrace::trace]
+    async fn transfer_ownership_to_admin(
+        &self,
+        object: &OwnershipObject,
+    ) -> databend_common_exception::Result<()> {
+        // Ensure accurate matching of a key
+        let match_seq = MatchSeq::GE(1);
+        let key = self.ownership_object_key(object);
+
+        let value = serialize_struct(
+            &OwnershipInfo {
+                object: object.clone(),
+                role: "account_admin".to_string(),
+            },
+            ErrorCode::IllegalUserInfoFormat,
+            || "",
+        )?;
+
+        let kv_api = self.kv_api.clone();
+        kv_api
+            .upsert_kv(UpsertKVReq::new(
+                &key,
+                match_seq,
+                Operation::Update(value),
+                None,
+            ))
+            .await?;
```

Comment:
Upsert returns two error: MetaError for service errorr, or it just failed to update any record if `match_seq` mismatches. 

Is it necessary to return an explicit error if nothing is updated? I.e., check the result `upsert_kv()` returns

Reply:
I think no need to check the err info. This transfer will only call when role be dropped and it owns object. If err we just need return err. And will not according to err kind do other logic. cc @flaneur2020

Reply:
The caller may choose to ignore the error, but for this function itself, it should be explicit and clear about what happened.

Reply:
imho we can append a context infomation in the error here, like "(while transfer_ownership_to_admin)"